KeywordSearch/src/org/sleuthkit/autopsy/keywordsearch/Server.java

```diff
@@ -2030,6 +2030,13 @@ private void connectToSolrServer(HttpSolrClient solrServer) throws SolrServerExc
         private final List<SolrInputDocument> buffer;
         private final Object bufferLock;
         
+        /* (JIRA-7521) Sometimes we get into a situation where Solr server is no longer able to index new data. 
+        * Typically main reason for this is Solr running out of memory. In this case we will stop trying to send new 
+        * data to Solr (for this collection) after certain number of consecutive batches have failed. */
+        private static final int MAX_NUM_CONSECUTIVE_FAILURES = 5;
+        private int numConsecutiveFailures = 0;
+        private boolean skipIndexing = false;
```

Comment:
@APriestman It should only be accessed by one thread at a time because the top level Server.addDocument(SolrInputDocument doc) has a lock. That's really the source of the entire Solr bottle neck. And since we are only incrementing the counter, the worst case scenario is that we will do an extra re-try before declaring the indexing to be "dead". But I'll add the "atomic" so that it's clearer, and because theoretically the SendBatchedDocumentsTask could access these variables.